library/src/scala/tasty/reflect/Kernel.scala

```diff
@@ -37,19 +37,19 @@ package scala.tasty.reflect
  *           |                             +- While
  *           |
  *           |
- *           +- TypeTree ----+- TypeTree_Inferred
- *           |               +- TypeTree_Ident
- *           |               +- TypeTree_Select
- *           |               +- TypeTree_Project
- *           |               +- TypeTree_Singleton
- *           |               +- TypeTree_Refined
- *           |               +- TypeTree_Applied
- *           |               +- TypeTree_Annotated
- *           |               +- TypeTree_MatchType
- *           |               +- TypeTree_ByName
- *           |               +- TypeTree_LambdaTypeTree
- *           |               +- TypeTree_TypeBind
- *           |               +- TypeTree_TypeBlock
+ *           +- TypeTree ----+- Inferred
+ *           |               +- TypeIdent
+ *           |               +- TypeSelect
+ *           |               +- Project
+ *           |               +- Singleton
+ *           |               +- Refined
+ *           |               +- Applied
+ *           |               +- Annotated
+ *           |               +- MatchTypeTree
+ *           |               +- ByName
+ *           |               +- LambdaTypeTree
+ *           |               +- TypeBind
+ *           |               +- TypeBlock
```

Comment:
This refactoring reminds me of a disagreement among @xeno-by @odersky @olafurpg and me about _whether type trees and types_ should be the same.

Now opinions seem to converge that they should not be the same, as it's implemented in Tasty Reflect. However, I think one issue left unaddressed: _the confusion to meta programmers why there are seemingly overlapping Types and TypeTrees_ and usability problems evidenced by this PR.

To address the usability issue, I propose to just have one abstract type `TypeTree` (which could be used for positions) and remove all APIs to extract or construct `TypeTree`. Instead, meta-programmers should always work with `Types`, which is more reliable, as we do in the compiler.

Just a thought for discussion, no need to address in this PR.